test/extended/etcd/leader_changes.go

```diff
@@ -0,0 +1,33 @@
+package etcd
+
+import (
+	"context"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"github.com/prometheus/common/model"
+
+	"github.com/openshift/origin/test/extended/prometheus/client"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("etcd", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLI("etcd-leader-change", exutil.KubeConfigPath())
+	)
+
+	g.It("leader changes are not excessive", func() {
+		prometheus, err := client.NewE2EPrometheusRouterClient(oc)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		g.By("Watching the total number of etcd leadership changes for the next five minutes")
+		// check sum(etcd_server_leader_changes_seen_total) every 30s for 5m.
+		// the value should consistently be less than 10.
+		o.Consistently(func() model.SampleValue {
+			result, err := prometheus.Query(context.Background(), "sum(etcd_server_leader_changes_seen_total)", time.Now())
```

Comment:
`max(etcd_server_leader_changes_seen_total)` should equal one.  Per a buried thread in the incident channel.

Reply:
Updated.

Reply:
It’s important to understand this is optimal (max leader change 1) and only true to my knowledge with AWS. So we will most likely fail on most clouds. I think we will need a knob to tune this per cloud. So we can adjust per cloud or only run on AWS to watch for regression.

Reply:
I need some guidance on what the numbers should be for other clouds. 

@wking, if you have a suggestion for azure, I could at least start with that.